Modify rule S5542[java]: Detect CBC mode when used with padding (APPSEC-31) (#1052)
SonarSource/rspec: rules/S5542/description.adoc

> Encryption operations should use a secure mode and padding scheme so that confidentiality and integrity can be guaranteed.

Reviewer: ```suggestion
Encryption algorithms should be used with a secure mode and padding scheme to guarantee data confidentiality and integrity.
```

* For block cipher encryption algorithms (like AES):

> ** The ECB (Electronic Codebook) cipher mode doesn't provide serious message confidentiality: under a given key any given plaintext block always gets encrypted to the same ciphertext block. This mode never be used.

Reviewer: ```suggestion
** The ECB (Electronic Codebook) cipher mode doesn't provide serious message confidentiality: under a given key any given plaintext block always gets encrypted to the same ciphertext block. This mode should never be used.
```
